src/views/User/User.tsx

```diff
@@ -119,8 +119,37 @@ export class User extends React.PureComponent<UserProperties, UserState> {
             this.setState({ user: undefined, resolved: true });
             return;
         }
+
+        // Cheaper API calls provide partial profile data before players/%%/full
+        Promise.all([get("players/%%", user_id), get("/termination-api/player/%%", user_id)])
+            .then((responses: [rest_api.PlayerDetail, rest_api.termination_api.Player]) => {
+                if (this.state.resolved) {
+                    return;
+                }
+                const user: rest_api.FullPlayerDetail["user"] = {
+                    ...responses[0],
+                    professional: responses[0].ui_class.indexOf("professional") >= 0,
+                    is_moderator: responses[0].ui_class.indexOf("moderator") >= 0,
+                    is_superuser: responses[0].ui_class.indexOf("admin") >= 0,
+                    is_tournament_moderator: false,
+                    is_watched: false,
+                    on_vacation: false,
+                    vacation_left: 0,
+                    deviation: responses[0].ratings.overall.deviation,
+                    ranking: responses[1].ranking,
+                    rating: responses[1].rating,
+                    ratings: responses[1].ratings,
+                    first_name: null,
+                    last_name: null,
```

Comment:
If `players/%%/full` completes before `players/%%` and `termination-api/player/%%`, won't these constants override the good values?

Reply:
If `full` returns first, `state.resolved` should be `true` right?

I check that at the beginning of this callback and exit early if that's the case.

Reply:
Right you are, I overlooked that, thanks!